File: web_shop_backend/src/main/java/com/lap/web_shop_backend/model/Product.java
package com.lap.web_shop_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String product_name;

    @Column(precision = 10, scale = 2)
    private BigDecimal price;

    private String description;

    private String image_path;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Product_property> properties;

    @OneToMany(mappedBy = "product")
    @JsonIgnore
    private List<Shipment_product> shipmentProducts;
}

File: web_shop_frontend/src/models/Product.ts
import Product_property from "./Product_property"

export default class Product {
    id: Number
    product_name: String
    price: Number
    description: String
    image_path: String
    properties: Product_property[]

    constructor(data: any) {
        this.id = data.id
        this.product_name = data.product_name
        this.price = data.price
        this.description = data.description
        this.image_path = data.image_path
        this.properties = new Array<Product_property>

        for(const property of data.properties)
            this.properties.push(new Product_property(property))
    }
}